best-practices/native/react-native/common-components/buttons/button.tsx

```diff
@@ -0,0 +1,78 @@
+import React from "react";
+import { StyleSheet, TouchableOpacity, Text, Image } from "react-native";
+import PropTypes from "prop-types";
+
+interface ButtonProps {
+  onPress: Function;
+  text: string;
+  icon: any;
+  height: number;
+  width: number;
+  backgroundColor: string;
+}
+
+type InternalOptions = {
+  height?: string;
+  width?: string;
+  backgroundColor?: string;
+  onPress?: Function;
+  text?: string;
+  icon: any;
+};
+
+const styles = StyleSheet.create({
+  button: {
+    display: "flex",
+    justifyContent: "center",
+    alignItems: "center",
+    borderRadius: 4,
+    marginTop: 5,
+    marginBottom: 5,
+  },
+  text: {
+    color: "#FFF",
+    fontFamily: "oswald",
+  },
+  icon: {
+    position: "absolute",
+    right: "85%",
+  },
+});
+
+const Button = ({ onPress, text, icon }: ButtonProps) => {
```

Comment:
If you want to type this component, I think you can do something like `React.FC<ButtonProps>`

Reply:
I've never seen this before! Super cooL! Thanks for your suggestions. All were noted and added!